plugins/send-by-email/plugin.vala

```diff
@@ -0,0 +1,132 @@
+/*
+* Copyright 2021 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+public class Files.Plugins.SendByEmailMenuItem : Gtk.MenuItem {
+    private GLib.File[] files;
+
+    public SendByEmailMenuItem (GLib.File[] files) {
+        this.files = files;
+
+        label = _("Send by Email");
+    }
+
+    public override void activate () {
+        try {
+            var portal = Portal.Email.get ();
+
+            window_export.begin ((obj, res) => {
+                var options = new HashTable<string, Variant> (str_hash, str_equal);
+                options["handle_token"] = Portal.generate_token ();
+
+                int[] file_descriptors = {};
+                foreach (var file in files) {
+                    file_descriptors += Posix.open (file.get_path (), Posix.O_RDONLY);
+                }
+                options["attachment_fds"] = file_descriptors;
```

Comment:
```suggestion
                var files_builder = new VariantBuilder (new VariantType ("ah"));
                var file_descriptors = new UnixFDList ();
                foreach (var file in files) {
                    var fd = Posix.open (file.get_path (), Posix.O_RDONLY | Posix.O_CLOEXEC);
                    if (fd == -1) {
                        warning ("send-by-mail: cannot open file: '%s'", file.get_path ());
                        continue;
                    }

                    files_builder.add ("h", file_descriptors.append (fd));
                }
                
                options["attachment_fds"] = files_builder.end ();
```
Here we create a UnixFDList and a VariantBuilder for the handles, we insert the fd in the list, and the returned index on the builder. 

Also, skip non-valid file descriptors.